packages/react-pages/src/node/dynamic-modules/PageStrategy.ts

```diff
@@ -2,199 +2,109 @@ import * as fs from 'fs-extra'
       extractStaticData,
-      loadPageData: defaultPageLoader,
-      addPageData,
       watchFiles,
+      ...apiForCustomSource,
     }
 
     // The file currently being processed.
-    let currentFile: File | null = null
+    // let currentFile: File | null = null
 
-    let pagesPromise = Promise.resolve(pageCache)
+    // let pagesPromise = Promise.resolve(pageCache)
     findPages(pagesDir, {
       ...helpers,
-      loadPageData: (file) => Promise.resolve(loadPageData(file, helpers)),
     })
 
-    this.getPages = () => pagesPromise
+    this.getPages = () => Promise.resolve(pagesDataKeeper.toPagesData())
     this.close = () => watchers.forEach((w) => w.close())
 
-    function addPageData({
-      pageId,
-      key = 'main',
-      dataPath,
-      staticData,
-    }: PageData) {
-      if (!pageId.startsWith('/'))
-        throw Error(
-          `addPageData error: pageId should start with "/", but got "${pageId}"`
-        )
-
-      const existingPage = pageCache[pageId]
-      const page =
-        existingPage ||
-        (pageCache[pageId] = {
-          data: {},
-          staticData: {},
-        })
-
-      if (dataPath) {
-        if (page.data[key])
-          throw Error(
-            `addPageData conflict: data with key "${key}" already exists`
-          )
-        page.data[key] = dataPath
-      }
-
-      if (staticData) {
-        if (page.staticData[key])
-          throw Error(
-            `addPageData conflict: staticData with key "${key}" already exists`
-          )
-        page.staticData[key] = staticData
-      }
-
-      if (currentFile) {
-        let dataKeys = currentFile.dataKeys.get(pageId)
-        if (!dataKeys) currentFile.dataKeys.set(pageId, (dataKeys = []))
-        dataKeys.push(key)
-      }
-
-      if (existingPage) {
-        changedPages.add(pageId)
-        emitChange()
-      }
-    }
-
-    function removePageData(dataKeys: string[], pageId: string) {
-      const page = pageCache[pageId]
-
-      dataKeys.forEach((key) => {
-        delete page.data[key]
-        delete page.staticData[key]
-      })
-
-      if (hasKeys(page.data) || hasKeys(page.staticData)) {
-        changedPages.add(pageId)
-        emitChange()
-      } else {
-        delete pageCache[pageId]
-      }
-    }
-
-    function hasKeys(obj: object) {
-      for (const _ in obj) return true
-      return false
-    }
-
-    function watchFiles(
-      baseDir: string,
-      arg2?: string | string[] | FileHandler,
-      arg3?: FileHandler
-    ) {
-      let globs: string[]
-      let handler: FileHandler
-
-      if (typeof arg2 === 'function') {
-        globs = ['**/*']
-        handler = arg2
-      } else {
-        globs = Array.isArray(arg2) ? arg2 : [arg2 || '**/*']
-        handler =
-          arg3 || ((file) => Promise.resolve(loadPageData(file, helpers)))
-      }
-
+    function watchFiles({
+      baseDir,
+      globs = ['**/*'],
+      fileHandler = defaultFileHandler,
+      ignored = [],
```

Comment:
I would prefer keeping `watchFiles` the way it was. If we want to support ignored globs, callers can use negated globs (eg: `!**/ignored`).

Reply:
Negated globs is not friendly to users. I think that's why `chokidar` provide a `ignore` option. By the way vite also take this way: 

https://github.com/vitejs/vite/blob/44294ba051a112847f4119c46c0e5dc400498241/packages/vite/src/node/server/index.ts#L274

Reply:
Not friendly? Do you mean less discoverable? If you make sure it's documented, that shouldn't be an issue.

Vite works like that because it doesn't let you define the watched globs. Instead, it watches the entire root. Since `watchFiles` receives a globs array, it's more elegant to use negated globs.

Reply:
You have a point. I will change that.

Reply:
Done.